cmd/node/node.go

```diff
@@ -749,19 +749,28 @@ func (app *App) initServices(
 		app.hare,
 		app.edVerifier,
 	)
-	fetcher.SetValidators(atxHandler, poetDb, proposalListener, blockHandler, proposalListener, app.txHandler, malfeasanceHandler)
+	fetcher.SetValidators(
+		fetch.ValidatorFunc(pubsub.DropPeerOnError(atxHandler.HandleAtxData, app.host, lg, pubsub.ValidationRejectErr)),
```

Comment:
can we maybe do DropPeerOnReject (as a wrapper around DropPeerOnError, if you want to keep that)?

Reply:
I thought it could be useful to make it more general, but If you don't think so I'll just make it `DropPeerOnValidationReject` and hardcode the error.

Reply:
i meant that you can keep both of them, but define "shortcut" for Reject, because it is used everywhere

Reply:
I din't feel like keeping the original implementation and the shortcut, it's easy enough to split out if we need to.  Having both added an extra layer of indirection that I felt was not necessary.

I would have been happy just with the more general version though, IMO the shortcut doesn't provide enough value to justify it's use over the more general version.

EG.

```
pubsub.DropPeerOnError(atxHandler.HandleAtxData, app.host, lg, pubsub.ValidationRejectErr)
```
VS
```
pubsub.DropPeerOnValidationRejectError(atxHandler.HandleAtxData, app.host, lg)
```

But I don't feel that strongly about this.